neo/Wallets/AssetDescriptor.cs

```diff
@@ -7,13 +7,15 @@ namespace Neo.Wallets
 {
     public class AssetDescriptor
     {
-        public UIntBase AssetId;
+        public object AssetId;
```

Comment:
Can we keep `UIntBase` but change everything to non-virtual? Or just change it to an interface?

Reply:
We can keep UIntBase or introduce IUInt interface but only shared contract between UInt160 and UInt256 is ToArray method. The rest methods are either type specific or covered by other interfaces (IEquatable, IComparable, ISerializable). I see the consumer of UIntBase/IUInt always has to cast it the underlying type to work with.

For AssetDescriptor class (and for TransferOutput class) we can come up with more descriptive design. The idea here is to move casting logic outside of the class.

Option #1:
```C#
    public class AssetDescriptor<TUInt>
    {
        public TUInt AssetId;
        public string AssetName;
        public byte Decimals;

        public AssetDescriptor(TUInt asset_id)
        {
                AssetId = asset_id
        }
```

Option #2:
```C#
    public class AssetDescriptor256 : IAssetDescriptor
    {
        public UInt256 AssetId;
        public string AssetName;
        public byte Decimals;

        public AssetDescriptor(UInt256 asset_id)
        {
                AssetId = asset_id
        }

        public byte[] IAssetDescriptor.GetAssestIdAsByteArray()
        {
            return AssetId.ToArray();
        }
```

I'd like to elaborate more on the subject but I couldn't have found any example AssetDescriptor class usage.

Reply:
The usage of `AssetDescriptor` is in [neo-gui](https://github.com/neo-project/neo-gui) and [neo-cli](https://github.com/neo-project/neo-cli).